it/src/test/java/com/google/cloud/teleport/it/spanner/DefaultSpannerResourceManagerTest.java

```diff
@@ -0,0 +1,312 @@
+
+  @Captor private ArgumentCaptor<Iterable<Mutation>> writeMutationCaptor;
+  @Captor private ArgumentCaptor<Iterable<String>> statementCaptor;
+  @Captor private ArgumentCaptor<String> instanceIdCaptor;
+  @Captor private ArgumentCaptor<String> databaseIdCaptor;
+
+  @Before
+  public void setUp() {
+    testManager = new DefaultSpannerResourceManager(spanner, testId, projectId, region);
+  }
+
+  @Test
+  public void testCreateTable_throwsExceptionWhenSpannerCreateInstanceFails()
+      throws ExecutionException, InterruptedException {
+    // arrange
+    when(spanner.getInstanceAdminClient().createInstance(any()).get())
+        .thenThrow(InterruptedException.class);
+    String statement =
+        "CREATE TABLE Singers (\n"
+            + "  SingerId   INT64 NOT NULL,\n"
+            + "  FirstName  STRING(1024),\n"
+            + "  LastName   STRING(1024),\n"
+            + ") PRIMARY KEY (SingerId)";
+
+    // act & assert
+    assertThrows(SpannerResourceManagerException.class, () -> testManager.createTable(statement));
+  }
+
+  @Test
+  public void testCreateTable_throwsExceptionWhenSpannerCreateDatabaseFails()
+      throws ExecutionException, InterruptedException {
+    // arrange
+    when(spanner.getDatabaseAdminClient().createDatabase(any(), any(), any()).get())
+        .thenThrow(InterruptedException.class);
+    String statement =
+        "CREATE TABLE Singers (\n"
+            + "  SingerId   INT64 NOT NULL,\n"
+            + "  FirstName  STRING(1024),\n"
+            + "  LastName   STRING(1024),\n"
+            + ") PRIMARY KEY (SingerId)";
+
+    // act & assert
+    assertThrows(SpannerResourceManagerException.class, () -> testManager.createTable(statement));
+  }
+
+  @Test
+  public void testCreateTable_throwsExceptionWhenSpannerUpdateDatabaseFails()
+      throws ExecutionException, InterruptedException {
+    // arrange
+    when(spanner.getDatabaseAdminClient().updateDatabaseDdl(any(), any(), any(), any()).get())
+        .thenThrow(InterruptedException.class);
+    String statement =
+        "CREATE TABLE Singers (\n"
+            + "  SingerId   INT64 NOT NULL,\n"
+            + "  FirstName  STRING(1024),\n"
+            + "  LastName   STRING(1024),\n"
+            + ") PRIMARY KEY (SingerId)";
+
+    // act & assert
+    assertThrows(SpannerResourceManagerException.class, () -> testManager.createTable(statement));
+  }
+
+  @Test
+  public void testCreateTable_worksWhenSpannerDoesntThrowAnyError() {
+    //   arrange
+    String statement =
+        "CREATE TABLE Singers (\n"
+            + "  SingerId   INT64 NOT NULL,\n"
+            + "  FirstName  STRING(1024),\n"
+            + "  LastName   STRING(1024),\n"
+            + ") PRIMARY KEY (SingerId)";
+
+    // act
+    testManager.createTable(statement);
+
+    // assert
+    verify(spanner.getDatabaseAdminClient())
+        .updateDatabaseDdl(
+            argThat(s -> s.startsWith("test-id") && !s.equals("test-id")),
```

Comment:
This should be more specific, since what follows "test-id" is important. With Truth, you can provide a regex that also matches the rest of the string.